extensions/binding/org.eclipse.smarthome.binding.coapconnectedlighting/src/main/java/org/eclipse/smarthome/binding/coapconnectedlighting/handler/CoAPConnectedLightingHandler.java

```diff
@@ -0,0 +1,361 @@
+    }
+
+    private void startBackgroundTemperatureRefresh() {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    getTemprature();
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            }
+        };
+
+        refreshTemperatureJob = scheduler.scheduleAtFixedRate(runnable, 0, refresh.intValue(), TimeUnit.SECONDS);
+    }
+
+    private void startBackgroundHumidityRefresh() {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    getHumidity();
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            }
+        };
+
+        refreshHumidityJob = scheduler.scheduleAtFixedRate(runnable, 0, refresh.intValue(), TimeUnit.SECONDS);
+    }
+
+    private void startBackgroundPressureRefresh() {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    getPressure();
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            }
+        };
+
+        refreshPressureJob = scheduler.scheduleAtFixedRate(runnable, 0, refresh.intValue(), TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (channelUID.getId().equals(CHANNEL_DimmableLight)) {
+                if (command instanceof PercentType) {
+                    PercentType commandString = ((PercentType) command);
+                    putlightDimmer(commandString.toString());
+                } else if (command instanceof DecimalType) {
+                    IncreaseDecreaseType commandString = ((IncreaseDecreaseType) command);
+                    putlightDimmer(commandString.toString());
+                    System.err.println(command.toString());
+                }
+            } else if (channelUID.getId().equals(CHANNEL_LCD)) {
+                if (command instanceof StringType) {
+                    StringType commandString = ((StringType) command);
+                    putLCD(commandString.toString());
+                }
+            } else if (channelUID.getId().equals(CHANNEL_WallSwitch)) {
+                if (command instanceof OnOffType) {
+                    OnOffType commandString = ((OnOffType) command);
+                    putLightOnOff(commandString.toString());
+                }
+            } else {
+                logger.error("Linked Channel does not support PUT. Invalid channel linked ");
+            }
+        } catch (Exception ex) {
```

Comment:
please narrow down the Exception to the types you expect here.

Reply:
@htreu This exception is pertinent to a communication error. It does not trigger in case of an invalid Channel type.